tools/hyperref-check.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -x
+
+# Extracting all broken hyperref detected by PDFTex
+for file in "./main/acle.md" "./morello/morello.md" "./mve_intrinsics/mve.md" "./neon_intrinsics/advsimd.md"; do
+  echo "Checking $file..."
+
+  pandoc $file --verbose --fail-if-warnings -o pdfs/tmp.pdf &> tools/erroutput.txt
+  cat tools/erroutput.txt | grep -E 'pdfTeX warning \(dest\): name{[^}]+}' | sed -E 's/.*name\{([^}]+)\}.*/\1/' | sort | uniq > tools/list_of_broken_links.txt
+
+  if [[ $(wc -l <<< tools/list_of_broken_links.txt) -gt 0 ]]; then
```

Comment:
I *think* you are using the wrong command:

```
17:21 $ echo $(wc -l <<< tools/list_of_broken_links.txt)
1
✔ ~/projects/acle [github.io-add-section-link-check|✚ 2…23⚑ 13]
17:22 $ wc -l tools/list_of_broken_links.txt
      81 tools/list_of_broken_links.txt
```

I think that what you want to use is to pipe the output of `cat` into `wc`:
```
✔ ~/projects/acle [github.io-add-section-link-check|✚ 2…23⚑ 13]
17:22 $ cat tools/list_of_broken_links.txt | wc -l
      81
✔ ~/projects/acle [github.io-add-section-link-check|✚ 2…23⚑ 13]
17:23 $ X=`cat tools/list_of_broken_links.txt | wc -l`
✔ ~/projects/acle [github.io-add-section-link-check|✚ 2…23⚑ 13]
17:23 $ echo $X
81 
```

Try to change the code as follows

```
number=`cat tools/list_of_broken_links.txt | wc -l`
if [[ "$number" -gt 0 ]] ; then
...
fi
```

Reply:
Check the previous commits. I already tried that. Not only did it not work, the CI was passing because it found no lines.

Reply:
Which one of the commits?

It works for me on my machine. This is the output store in a file `x` via `./tools/hyperref-check.sh &> x`:

```
+ for file in '"./main/acle.md"' '"./morello/morello.md"' '"./mve_intrinsics/mve.md"' '"./neon_intri\
nsics/advsimd.md"'
+ echo 'Checking ./main/acle.md...'
Checking ./main/acle.md...
+ pandoc ./main/acle.md --verbose --fail-if-warnings -o pdfs/tmp.pdf
+ cat tools/erroutput.txt
+ grep -E 'pdfTeX warning \(dest\): name{[^}]+}'
+ sed -E 's/.*name\{([^}]+)\}.*/\1/'
+ sort
+ uniq
++ cat tools/list_of_broken_links.txt
++ wc -l
+ X='      81'
+ [[       81 -gt 0 ]]
+ echo '**** WARNING! These section links have been found not to be working: '
**** WARNING! These section links have been found not to be working:
+ cat tools/list_of_broken_links.txt
sec-Attributes-and-pragmas
sec-Constant-arguments-to-intrinsics
sec-Feature-test-macros
sec-MVE-intrinsics
sec-NEON-intrinsics
sec-Parallel-selection
sec-TME-intrinsics
sec-Weak-linkage
ssec-32SIMD
ssec-AES
...
...
...
ssec-rng
ssec-swap
ssec-sysreg
ssec-v8max
ssec-v8rnd
ssec-vectypes
+ exit 1
```

These are the changes I applied to make it work: 

```
diff --git a/tools/hyperref-check.sh b/tools/hyperref-check.sh
index 3c99ae2..5081042 100755
--- a/tools/hyperref-check.sh
+++ b/tools/hyperref-check.sh
@@ -8,7 +8,8 @@ for file in "./main/acle.md" "./morello/morello.md" "./mve_intrinsics/mve.md" ".
   pandoc $file --verbose --fail-if-warnings -o pdfs/tmp.pdf &> tools/erroutput.txt
   cat tools/erroutput.txt | grep -E 'pdfTeX warning \(dest\): name{[^}]+}' | sed -E 's/.*name\{([^}]+)\}.*/\1/' | sort | uniq > tools/list_of_broken_links.txt

-  if [[ $(wc -l <<< tools/list_of_broken_links.txt) -gt 0 ]]; then
+  X=`cat tools/list_of_broken_links.txt | wc -l`
+  if [[ "$X" -gt 0 ]]; then
       echo "**** WARNING! These section links have been found not to be working: "
       cat tools/list_of_broken_links.txt
       exit 1
```

Reply:
It's in this commit
https://github.com/ARM-software/acle/pull/112/commits/ae8aa674a6f55aacc11df38bc0ad3c18b304434f
Also I wouldn't take a change working locally as an indication that the CI will function properly.
Almost every commit so far had been tested locally and worked fine, but the CI build didn't work in the same way.